pkg/workceptor/kubernetes.go

```diff
@@ -224,6 +236,106 @@ func podRunningAndReady(kw KubeUnit) func(event watch.Event) (bool, error) {
 	return inner
 }
 
+// podInfrastructureFailure checks if the pod has failed to start due to infrastructure issues.
+func (ku KubeAPIWrapper) podInfrastructureSuccess(pod *corev1.Pod) (bool, string, error) {
+	if pod == nil {
+		return false, "pod is nil", fmt.Errorf("pod is nil")
+	}
+
+	switch pod.Status.Phase {
+	case corev1.PodFailed:
+		for _, cs := range pod.Status.ContainerStatuses {
+			if cs.State.Terminated != nil {
+				switch cs.State.Terminated.Reason {
+				case "OOMKilled", "Evicted":
+					return false, cs.State.Terminated.Reason, nil
```

Comment:
```suggestion
		for _, cs := range pod.Status.ContainerStatuses {
			if cs.State.Terminated != nil {
				switch cs.State.Terminated.ExitCode {
				case 0:
				        break
				default:
					return false, cs.State.Terminated.Reason, nil
```

We can simplify on `cs.State.Terminated.ExitCode` because an exit code of 0 is success and everything else is failure.